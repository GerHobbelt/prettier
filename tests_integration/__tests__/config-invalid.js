"use strict";

const runPrettier = require("../runPrettier");

expect.addSnapshotSerializer(require("../cwd-serializer"));

test("throw error with invalid config format", () => {
  const output = runPrettier("cli/config/invalid", [
    "--config",
    "file/.prettierrc"
  ]);
  expect(output.stderr).toMatchSnapshot();
  expect(output.status).not.toBe(0);
});

test("throw error with invalid config target (directory)", () => {
  const output = runPrettier("cli/config/invalid", [
    "--config",
    "folder/.prettierrc" // this is a directory
  ]);
  expect(output.stderr).toMatchSnapshot();
  expect(output.status).not.toBe(0);
});

test("throw error with invalid config option (int)", () => {
  const output = runPrettier("cli/config/invalid", ["--config", "option/int"]);
  expect(output.stderr).toMatchSnapshot();
  expect(output.status).not.toBe(0);
});

test("throw error with invalid config option (trailingComma)", () => {
  const output = runPrettier("cli/config/invalid", [
    "--config",
    "option/trailingComma"
  ]);
  expect(output.stderr).toMatchSnapshot();
  expect(output.status).not.toBe(0);
});
