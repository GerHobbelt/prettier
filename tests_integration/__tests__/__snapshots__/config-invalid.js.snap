// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throw error with invalid config format 1`] = `
"Error: Invalid configuration file.
Failed to parse \\"<cwd>/tests_integration/cli/config/invalid/file/.prettierrc\\" as JSON, JS, or YAML.
"
`;

exports[`throw error with invalid config option (int) 1`] = `
"Error: Invalid --tab-width value.
Expected an integer, but received: 0.5
"
`;

exports[`throw error with invalid config option (trailingComma) 1`] = `
"Error: Invalid option for --trailing-comma.
Expected \\"none\\", \\"es5\\" or \\"all\\", but received: \\"wow\\"
"
`;

exports[`throw error with invalid config precedence option (configPrecedence) 1`] = `
"Error: Invalid option for --config-precedence.
Expected \\"cli-override\\", \\"file-override\\" or \\"prefer-file\\", but received: \\"option/configPrecedence\\"
"
`;

exports[`throw error with invalid config target (directory) 1`] = `
"Error: Invalid configuration file.
EISDIR: illegal operation on a directory, read
"
`;
